testing/common_testing_util.py

```diff
@@ -107,3 +121,96 @@ def printv(string, verbose=1, type='normal'):
                 print(string)
         except Exception:
             print(string)
+
+
+def assert_empty_bids_dataframe(loader_parameters: dict):
+    """Assertion function used during unit test to check if the loader parameters result in an empty bids dataframes.
+    Used as part of pytest-case scenarios for when the Step1/2/3 filters excluded almost all data. (e.g. when no
+    participants have the required Session data)
+
+    Args:
+        loader_parameters (dict): a dictionary object containing all loader parameters necessary derived from
+        config.json to construct the BidsDataFrame
+    """
+
+    # Create the bids frame.
+    bids_df = BidsDataframe(loader_parameters,
+                            str(path_data_multi_sessions_contrasts_tmp),
+                            derivatives=True)
+
+    # Assert new dataframe EQUALs an empty pandas dataframe
+    assert bids_df.df.equals(pd.DataFrame())
+
+
+def bids_dataframe_comparison_framework(loader_parameters: dict, target_csv: str):
+    """Main test function used to set up a CSV comparison between expected vs the output from the test
+    Args:
+        loader_parameters (dict): a dictionary object containing all loader parameters necessary derived from
+        config.json to construct the BidsDataFrame
+
+        target_csv (str): the filename string  of a CSV which indicate the goal of the unit test scenario. It is
+        derived from the bids_dataframe after stripping out its PATH column.
+    """
+    # Create the bids frame.
+    bids_df = BidsDataframe(loader_parameters,
+                            str(path_data_multi_sessions_contrasts_tmp),
+                            derivatives=True)
+
+    # Drop path as that can varies across runs.
+    df_test = bids_df.df.drop(columns=[BidsDataFrameKW.PATH])
+
+    # Sorting to ensure consistencies when comparing with ground truth.
+    df_test = df_test.sort_values(by=[BidsDataFrameKW.FILENAME]).reset_index(drop=True)
+
+    # Generate full path to the target reference that we hope to match (e.g. df_ref.csv or any other scenario specific
+    # CSV file name
+    csv_ref = os.path.join(loader_parameters[LoaderParamsKW.PATH_DATA][0], target_csv)
+
+    # Generate full path for the csv which is produced from Bids Dataframe so that we can save the CSV to that location
+    csv_test = os.path.join(loader_parameters[LoaderParamsKW.PATH_DATA][0], "df_test.csv")
```

Comment:
Just wondering why we use `os.path.join` instead of `Path()` here?

Reply:
Good point. Rewrite via PATH